lib/trino-record-decoder/src/test/java/io/trino/decoder/avro/TestAvroDecoder.java

```diff
@@ -369,6 +373,14 @@ public void testSchemaEvolutionToIncompatibleType()
                 .hasMessageMatching("Decoding Avro record failed.");
     }
 
+    @Test
+    public void testLongDecodedAsTimestampMillis()
+    {
+        DecoderTestColumnHandle row = new DecoderTestColumnHandle(0, "row", TIMESTAMP_MILLIS, "id", null, null, false, false, false);
+        Map<DecoderColumnHandle, FieldValueProvider> decodedRow = buildAndDecodeColumn(row, "id", "\"long\"", 1658463479L);
```

Comment:
Should we specify the `LogicalType` here

Reply:
See also:

> @hashhar You can read this commit https://github.com/trinodb/trino/commit/52f4cf28912f5e2d5e4933464a0a6a39a04e1bfa that I revert trino-kafka adapting Avro logical type.

The convertor logic will be implemented on demand from the plugin - Pulsar Trino plugin implement it ([code](https://github.com/apache/pulsar/blob/9529850bd48557dcba124a157a69b75b8f41da3b/pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/avro/PulsarAvroRowDecoderFactory.java#L119)).

Reply:
This code maps the Logical type to Trino type - which is correct. But `buildAndDecodeColumn` creates a schema without any logical type being annotated

Reply:
You're right. Let me try to extend the test utils.